src/util/file_download.ts

```diff
@@ -0,0 +1,27 @@
+import { isAndroid } from "./is_android";
+import { isSafari } from "./is_safari";
+
+export const fileDownload = (
+  element: HTMLElement,
+  href: string,
+  filename: string
+): void => {
+  const a = document.createElement("a");
+  a.target = "_blank";
+  a.href = href;
+  a.download = filename;
+
+  element.shadowRoot!.appendChild(a);
+
+  a.addEventListener("click", () => {
+    element.shadowRoot!.removeChild(a);
+  });
+
+  if (isSafari || isAndroid) {
+    // Directy calling a.click() is not working in mobile, fireing a MouseEvent is used as a workaround.
+    // https://github.com/home-assistant/frontend/issues/9374
+    a.dispatchEvent(new MouseEvent("click"));
```

Comment:
Weird... `click()` should do that... whatevah 🤷‍♂️

Reply:
ideally yes, but this feels less of a hack than the delay, and it has no delay :P

Reply:
Btw, so this doesn't work for desktop?

Reply:
It might, in my testing it did, but better to keep that as is to not get any surprises?

Reply:
I don't know, if it saves use checking useragents 🤷🏻‍♂️

Can we find some docs on why it behaves different on mobile?

Reply:
That would be great!
I have not found _anything_ that would suggest it should (other than the fact that it's not the same)

Reply:
Tested `a.dispatchEvent(new MouseEvent("click"))` with:
- Chrome/Firefox (desktop)
- Safari (iOS)
- Chrome/Safari (MacOS)

Should be fairly safe to switch to it.